jdbcSource/src/test/java/io/vantiq/extsrc/jdbcSource/TestJDBC.java

```diff
@@ -540,6 +555,95 @@ public void testDBReconnect() throws VantiqSQLException {
+        assumeTrue(testAuthToken != null && testVantiqServer != null);
+        assumeTrue(testDBUsername != null && testDBPassword != null && testDBURL != null && jdbcDriverLoc != null);
+        
+        // Check that Source does not already exist in namespace, and skip test if it does
+        assumeFalse(checkSourceExists());
+        
+        jdbc.setupJDBC(testDBURL, testDBUsername, testDBPassword);
+        
+        // Setup a VANTIQ JDBC Source, and start running the core
+        setupSource(createSourceDef());
+        
+        // Publish to the source in order to create a table
+        Map<String,Object> create_params = new LinkedHashMap<String,Object>();
+        create_params.put("query", CREATE_TABLE_MAX_MESSAGE_SIZE);
+        vantiq.publish("sources", testSourceName, create_params);
+        
+        // Insert 2000 rows into the the table
+        Map<String,Object> insert_params = new LinkedHashMap<String,Object>();
+        insert_params.put("query", INSERT_ROW_MAX_MESSAGE_SIZE);
+        for (int i = 0; i < 2000; i ++) {
+            vantiq.publish("sources", testSourceName, insert_params);
+        }
+        
+        // Query the Source without bundleFactor and make sure there were no errors
+        Map<String,Object> params = new LinkedHashMap<String,Object>();
+        params.put("query", QUERY_TABLE_MAX_MESSAGE_SIZE);
+        VantiqResponse response = vantiq.query(testSourceName, params);
+        JsonArray responseBody = (JsonArray) response.getBody();
+        assert responseBody.size() == 2000;
+        assert core.lastRowBundle.length == JDBCCore.DEFAULT_BUNDLE_SIZE;
+        
+        // Query with an invalid bundleFactor
+        params.put("bundleFactor", "jibberish");
+        response = vantiq.query(testSourceName, params);
+        responseBody = (JsonArray) response.getBody();
+        assert responseBody.size() == 2000;
+        assert core.lastRowBundle.length == JDBCCore.DEFAULT_BUNDLE_SIZE;
+        
+        // Query with an invalid bundleFactor
+        params.put("bundleFactor", -1);
+        response = vantiq.query(testSourceName, params);
+        responseBody = (JsonArray) response.getBody();
+        assert responseBody.size() == 2000;
+        assert core.lastRowBundle.length == JDBCCore.DEFAULT_BUNDLE_SIZE;
+        
+        // Query with bundleFactor that divides evenly into 2000 rows
+        params.put("bundleFactor", 500);
+        response = vantiq.query(testSourceName, params);
+        responseBody = (JsonArray) response.getBody();
+        assert responseBody.size() == 2000;
+        assert core.lastRowBundle.length == 500;
+        
+        // Query with bundleFactor that doesn't divide evenly into 2000 rows
+        params.put("bundleFactor", 600);
+        response = vantiq.query(testSourceName, params);
+        responseBody = (JsonArray) response.getBody();
+        assert responseBody.size() == 2000;
+        assert core.lastRowBundle.length == 200;
```

Comment:
can probably do the remainders (both) as 
```
core.lastRowBundle.length = ( rowCount % bundleSize)
```
where rowCount is a variable set to, in this case 2000, and bundleSize is a variable set to 500.

Indeed, both of these can be in a helper that has those as parameters.  Since only 2, not really necessary, but to consider in the future.